### PR TITLE
Release 2.6

### PR DESCRIPTION
**Pull request by StorytellerCZ**
* Remove deprecated APIs
* Add support for arbitrary iterables in #each templates

### CODE REVIEW

**Comment by StorytellerCZ** [comment]
My apologies for the crazy commits, had some issues with Git due to setting once upstream branch wrong. Should be all fixed now.

**Review by filipenevola — CHANGES_REQUESTED** [review]
Include this new package in the publish.sh script

**Comment by jankapunkt** [comment]
@StorytellerCZ what's the status here / is there something todo left?

**Comment by StorytellerCZ** [comment]
I would like to include #358 into this as well as the final piece. Also waiting for approval from @filipenevola to proceed.

**Comment by fredmaiaarantes** [comment]
@StorytellerCZ @jankapunkt, I have updated the test-app to Meteor 2.7 and all the tests are ok.

From your point of view this PR is ready to be merged? If so, I can do it today.

**Comment by jankapunkt** [comment]
I can't review before Monday but then I would like to do so.

**Comment by fredmaiaarantes** [comment]
> I can't review before Monday but then I would like to do so. 

No problem, you can just ping me.

**Comment by StorytellerCZ** [comment]
Let's include #358 into this as well.

**Comment by jankapunkt** [comment]
Agree 👍

File: packages/observe-sequence/observe_sequence_tests.js
Comment (jankapunkt) [inline packages/observe-sequence/observe_sequence_tests.js:220]: Should we add a few more tests here? I think if we publish this and productions will use this there may be unforseen edge-cases.

I would propose to add tests for:

- [x] empty map
- [ ] arbitrary filled map
- [ ] empty set
- [x] arbitrary filled set
- [ ] empty custom structure implementing the iterable protocol
- [ ] arbitrary filled custom structure implementing the iterable protocol

File: packages/spacebars/spacebars-runtime.js
Comment (jankapunkt) [inline packages/spacebars/spacebars-runtime.js:283]: We should add a note to the changelog/history,  that this API is removed

File: packages/templating-runtime/templating.js
Comment (jankapunkt) [inline packages/templating-runtime/templating.js:40]: We should add a note to the changelog/history,  that this API is removed

File: packages/templating-runtime/templating.js
Comment (jankapunkt) [inline packages/templating-runtime/templating.js:186]: We should add a note to the changelog/history,  that this API is removed

**Review by jankapunkt — CHANGES_REQUESTED** [review]
I think we should do a last round and add a few more tests for the new iterable feature to prevent Blaze users become unintended testers and add notes to the breaking changes in README or Changelog.

File: packages/observe-sequence/observe_sequence_tests.js
Comment (StorytellerCZ) [reply packages/observe-sequence/observe_sequence_tests.js:220]: More tests are always welcome.

**Comment by StorytellerCZ** [comment]
I'll add notice about the removed APIs to history.

**Comment by fredmaiaarantes** [comment]
As soon as the comments are addressed, we can merge this PR.

File: HISTORY.md
Comment (StorytellerCZ) [reply HISTORY.md]: Done!

File: packages/blaze-html-templates/package.js
Comment (StorytellerCZ) [reply packages/blaze-html-templates/package.js:18]: added

**Review by jankapunkt — APPROVED** [review]
Generally approved, only a minor change could be applied, which I personally think could be kept due to fun reasons :-D

File: HISTORY.md
Comment (StorytellerCZ) [reply HISTORY.md:8]: That is a good catch :laughing: :beers: Would be a nice experiment to leave it there to see who read the changelogs.

**Review by fredmaiaarantes — APPROVED** [review]

**Comment by denihs** [comment]
Guys, everything is now published. Awesome work!

**Comment by StorytellerCZ** [comment]
@denihs great work! Looks like the docs page still needs to be manually updated.

**Pull request merged**
